modules/dnn/src/layers/reduce2_layer.cpp

```diff
@@ -0,0 +1,568 @@
+                        accumulator.update(static_cast<WT>(loop_p_src[l]));
+                    }
+                }
+                auto val = accumulator.finalize();
+                p_dst[i] = saturate_cast<dtype>(static_cast<double>(val));
+
+                ++loop;
+                if (loop >= last_unreduced_dim) {
+                    loop = 0;
+                    ++main_index;
+                    if (main_index < unprojected_steps.size())
+                        origin = unprojected_steps[main_index];
+                } else {
+                    origin += last_unreduced_step;
+                }
+            }
+        }
+    };
+
+    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+
+        CV_Assert(!inputs.empty());
+        Mat& src = inputs[0];
+        std::vector<int> axes;
+        if (!this->axes.empty()) {
+            axes = this->axes;
+        } else if (inputs.size() >= 2) {
+            tensorToIntVec(inputs[1], axes);
+        }
+
+        MatShape inpShape = shape(src);
+        MatShape outShape;
+        if (axes.empty()) {
+            if (noop_with_empty_axes) {
+                outShape = inpShape;
+            } else {
+                if (keepdims) {
+                    outShape = inpShape;
+                    for (int i = 0; i < (int)outShape.size(); ++i) outShape[i] = 1;
+                } else {
+                    outShape.assign(1, 1);
+                }
+            }
+        } else {
+            std::vector<int> norm_axes = axes;
+            for (size_t i = 0; i < norm_axes.size(); ++i)
+                norm_axes[i] = normalize_axis(norm_axes[i], inpShape);
+            MatShape tmp = inpShape;
+            for (int a : norm_axes) tmp[a] = -1;
+            for (size_t i = 0; i < tmp.size(); ++i) {
+                if (tmp[i] == -1) {
+                    if (keepdims) outShape.push_back(1);
+                } else {
+                    outShape.push_back(tmp[i]);
+                }
+            }
+            if (outShape.empty()) outShape.push_back(1);
+            axes = norm_axes;
+        }
+
+        auto kind = outputs_arr.kind();
+        if (kind == _InputArray::STD_VECTOR_MAT) {
+            outputs_arr.getMatVecRef()[0].fit(outShape, src.type());
+        } else {
+            CV_Assert(kind == _InputArray::STD_VECTOR_UMAT);
+            outputs_arr.getUMatVecRef()[0].fit(outShape, src.type());
+        }
+        outputs_arr.getMatVector(outputs);
+        Mat& dst = outputs[0];
+
+        typeDispatch(dst.type(), src, dst, axes, noop_with_empty_axes);
+    }
+
+    virtual std::ostream& dumpAttrs(std::ostream& strm, int indent) const CV_OVERRIDE
+    {
+        prindent(strm, indent);
+        strm << "reduce_type: \"";
```

Comment:
replace it with `strm << "reduce_type: \"" << reduceTypeToString(reduce_type) << "\",\n";`